src/core_atmosphere/physics/mpas_atmphys_control.F

```diff
@@ -493,6 +495,79 @@ subroutine physics_compatibility_check(dminfo, blockList, streamManager, ierr)
 
  end subroutine physics_compatibility_check
 
+!=================================================================================================================
+!  routine sst_update_check()
+!
+!> \brief Sea Surface Interval Update Check
+!> \author Miles Curry
+!> \date   January 2019
+!> \details
+!>  This routine checks that the user has correctly set the SST update options,
+!>  by setting the `config_sst_update` namelist option equal to `true` .AND.
+!>  by setting the `input_interal` surface stream variable in streams.atmosphere
+!>  to a valid HH:MM:SS update time.
+!>
+ subroutine sst_update_check(blockList, streamManager, ierr)
+!=================================================================================================================
+
+ use mpas_timekeeping, only : mpas_split_string
+
+ implicit none
+
+ ! Input Variables
+ type (block_type), pointer :: blockList
+ type (MPAS_streamManager_type), pointer :: streamManager
+ integer, intent(out) :: ierr
+
+ ! Local Variables
+ logical,pointer:: config_sst_update
+ character(len=StrKIND) :: stream_interval
+
+
+ call mpas_pool_get_config(blocklist % configs, 'config_sst_update', config_sst_update)
+ call MPAS_stream_mgr_get_property(streamManager, 'surface', MPAS_STREAM_PROPERTY_RECORD_INTV, stream_interval, &
+                                   direction=MPAS_STREAM_INPUT, ierr=ierr)
+
+ write(0,*) "stream_interval is: ", stream_interval
+ write(0,*) "config_sst_update is: ", config_sst_update
+
+
+ if ( config_sst_update .AND. stream_interval == 'none' ) then
+    call mpas_log_write('*******************************************************************************', &
+                        messageType=MPAS_LOG_ERR)
+    call mpas_log_write("'config_sst_update' in the physics section of the namelist.atmosphere file was set", &
+                    messageType=MPAS_LOG_ERR)
+    call mpas_log_write("to 'true', but the 'input_interval' in the surface stream of the streams.atmosphere", &
+                    messageType=MPAS_LOG_ERR)
+    call mpas_log_write("file has not been set.", &
+                    messageType=MPAS_LOG_ERR)
+    call mpas_log_write("Please set the update interval to a valid DD_HH:MM:SS interval or turn 'config_sst_update'", &
+                    messageType=MPAS_LOG_ERR)
+    call mpas_log_write("to 'false'.", &
+                    messageType=MPAS_LOG_ERR)
+    call mpas_log_write('*******************************************************************************', &
+                        messageType=MPAS_LOG_ERR)
+    call mpas_log_write("FATAL ERROR: SST Interval Update Error - see above", &
+                        messageType=MPAS_LOG_CRIT)
+    ierr = ierr + 1
```

Comment:
Because `ierr` is declared as `intent(out)`, I think the compiler is allowed to let the variable be set to undefined values upon entry to the subroutine (with the expectation that it will always be assigned a value. As such, this statement may give `ierr` an undefined value. It appears that the `physics_compatibility_check` routine may have the same problem (but that is likely my fault!). It should be safe here to simply set `ierr` to 1.

Reply:
Update: I see that `ierr` is initialized to zero in the `physics_compatibility_check`, so there's no problem there.